src/PluginsManager/Microsoft.Performance.Toolkit.PluginsManager.Core/AvailablePlugin.cs

```diff
@@ -20,14 +20,16 @@ public sealed class AvailablePlugin
             string displayName,
             string description,
             Uri packageUri,
-            Guid fetcherTypeId) 
+            Guid fetcherTypeId,
+            IPluginDiscoverer discoverer) 
```

Comment:
If we have the pluginDiscoverer, instead of having the person instantiating AvailablePlugin finding the pluginSource and the URI, should we extrapolate that information here in the ctor?

Is it possible for the package URI and PluginSource to be different from the ones on the IPluginDiscoverer?

Reply:
Good call out. Will extrapolate `pluginSource` from the discoverer. The `packageUri` should be specified by the caller though, as it's pointing to where the plugin package should be downloaded from.

Reply:
See https://github.com/microsoft/microsoft-performance-toolkit-sdk/pull/253 for the alternative approach to solve this problem.